libraries/Microsoft.Bot.Builder.Azure/Storage/Blobs.cs

```diff
@@ -0,0 +1,243 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web;
+using Azure;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Azure.Storage
+{
+    /// <summary>
+    /// Implements <see cref="IStorage"/> using Azure Storage Blobs.
+    /// </summary>
+    /// <remarks>
+    /// This class uses a single Azure Storage Blob Container.
+    /// Each entity or <see cref="IStoreItem"/> is serialized into a JSON string and stored in an individual text blob.
+    /// Each blob is named after the store item key,  which is encoded so that it conforms a valid blob name.
+    /// If an entity is an <see cref="IStoreItem"/>, the storage object will set the entity's <see cref="IStoreItem.ETag"/>
+    /// property value to the blob's ETag upon read. Afterward, an <see cref="BlobRequestConditions"/> with the ETag value
+    /// will be generated during Write. New entities start with a null ETag.
+    /// </remarks>
+    public class Blobs : IStorage
+    {
+        private static readonly JsonSerializer JsonSerializer = JsonSerializer.Create(new JsonSerializerSettings
+        {
+            // we use All so that we get typed roundtrip out of storage, but we don't use validation because we don't know what types are valid
+            TypeNameHandling = TypeNameHandling.All,
+        });
+
+        // If a JsonSerializer is not provided during construction, this will be the default static JsonSerializer.
+        private readonly JsonSerializer _jsonSerializer;
```

Comment:
>_jsonSerializer [](start = 40, length = 15)

Having a private field _jsongSerializer and a static field JsonSerializer is confusing, can we use just one? In the ode below we sometimes use _jsongSerializer and sometimes we use JsonSerializer